src/redux/books/books.js

```diff
@@ -1,48 +1,94 @@
-import { v4 as uuidv4 } from 'uuid';
-
+const CONSUME_API_BEGIN = 'antiquarian/books/CONSUME_API_BEGIN';
+const GET_ERROR = 'antiquarian/books/GET_ERROR';
 const ADD_BOOK = 'antiquarian/books/ADD_BOOK';
+const GET_BOOK = 'antiquarian/books/GET_BOOK';
 const REMOVE_BOOK = 'antiquarian/books/REMOVE_BOOK';
 
-const addBook = (payload) => ({
+// Loading action creator
+const consumeApiBegin = () => ({
+  type: CONSUME_API_BEGIN,
+  payload: { loading: false },
+});
+
+// Error action creator
+const getError = (payload) => ({
+  type: GET_ERROR,
+  payload,
+});
+// Make API call to add a new book to database
+export const addBook = (payload) => ({
   type: ADD_BOOK,
   payload,
 });
 
-const removeBook = (payload) => ({
+// Make API call to get books from database
+export const getBook = (payload) => ({
+  type: GET_BOOK,
+  payload,
+});
+
+// Make API call to remove book from database
+export const removeBook = (payload) => ({
   type: REMOVE_BOOK,
   payload,
 });
 
-const initialState = [
-  {
-    id: uuidv4(),
-    title: 'The Hunger Games',
-    author: 'Suzanne Collins',
-  },
-  {
-    id: uuidv4(),
-    title: 'Dune',
-    author: 'Frank Herbert',
-  },
-  {
-    id: uuidv4(),
-    title: 'Capital in the Twenty-First Century',
-    author: 'Suzanne Collins',
-  },
-];
+// Create action that returns function to handle API calls
+export const consumeApi = (params) => ((dispatch) => {
+  const {
+    method, body, itemId, actionCreator,
+  } = params;
+  const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/tWY3c29a1dHLDQJJS36r/books';
+  // show loading text
+  dispatch(consumeApiBegin());
+  // fetch based on parameters
+  fetch(`${baseUrl}/${itemId}`, {
+    method,
+    body,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    // Necessary because get returns json.
+    // Absence of condition causes the json.parse error line 1 column 1
+  }).then((response) => ((method === 'GET') ? response.json() : response.text()))
+    .then((response) => {
+      const data = response;
+      // Either getBooks, addBooks or deleteBooks
+      dispatch(actionCreator(data));
+    }).catch((error) => {
+      const errorMessage = error.message;
+      dispatch(getError(errorMessage));
+    });
+}
+);
```

Comment:
- Consider letting the reducer in one component to avoid a lot of code in just one component it would look more clean and easy to distinguish from other code inside. Great work!

Reply:
Thanks, man. I will take note of that.